engine/src/main/java/io/graphqlcrud/GraphQLSchemaBuilder.java

```diff
@@ -195,15 +257,35 @@ private void addQueryOperationsForEntity(Entity entity, Builder queryTypeBuilder
         return fields;
     }
 
-    protected List<GraphQLInputObjectField.Builder> buildFilterFields(Entity filterEntity, Schema schema) {
+    protected List<GraphQLInputObjectField.Builder> buildFilterFields(Entity filterEntity) {
         ArrayList<GraphQLInputObjectField.Builder> filterFields = new ArrayList<GraphQLInputObjectField.Builder>();
 
         filterEntity.getAttributes().stream().forEach(attr -> {
             GraphQLInputObjectField.Builder builder = GraphQLInputObjectField.newInputObjectField();
             builder.name(attr.getName());
-            builder.type(TYPEMAP.getAsGraphQLFilterType(attr.getType()));
+            builder.type(TYPE_MAP.getAsGraphQLTypeForInput(attr.getType(), "filter"));
             filterFields.add(builder);
         });
         return filterFields;
     }
+
+    protected List<GraphQLInputObjectField.Builder> buildMutations(Entity mutationEntity) {
+        ArrayList<GraphQLInputObjectField.Builder> mutationFields = new ArrayList<>();
+
+        mutationEntity.getAttributes().forEach(attribute -> {
+            GraphQLInputObjectField.Builder builder = GraphQLInputObjectField.newInputObjectField();
+            builder.name(attribute.getName());
+            if (mutationEntity.isPartOfPrimaryKey(attribute.getName())) {
+                builder.type(GraphQLNonNull.nonNull(Scalars.GraphQLID));
+            } else {
+                if (attribute.isNullable()) {
+                    builder.type(TYPE_MAP.getAsGraphQLTypeForInput(attribute.getType(), "mutation"));
```

Comment:
Looks like there is `filter` input type on `update` and should be there also on `delete`

Reply:
> Looks like there is `filter` input type on `update` and should be there also on `delete`

`filter input type`? Do you mean separate `input type` for delete? 

For both update and delete we have the same `Mutate<tablename>Input` input type.

Reply:
https://graphqlcrud.org/docs/next/update#conditional-updates